_adr/87/index.adoc

```diff
@@ -1,48 +1,453 @@
+** [.underline]#Definition of Done#:
+*** New customer facing metric should be included in the response of the following KAS Fleet Manager API endpoints:
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByInstantQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query]
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByRangeQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query_range]
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/federateMetrics[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/federate]
+
+. Document the new metric
+** This should be assigned to the Documentation/Customer Content Support team.
+** [.underline]#Definition of Done#:
+*** New metric documented in the app-services-guides: https://github.com/redhat-developer/app-services-guides/tree/main/docs/kafka/metrics-monitoring-kafka[Monitoring metrics in OpenShift Streams for Apache Kafka].
+
+
+### Migrating existing metrics to the new naming policy
+KAS Fleet Manager does not support API versioning at the moment so we would need to support both existing and new metrics during the migration process.
+
+The Kafka console provides dashboards which can display data up to the last 7 days (see the design in https://redhat-developer.github.io/app-services-ui-components/?path=/story/kafka-metrics--all-ready[Storybook]).
+Since the revised metric names will be added as new metrics, they will have no history.
+However, there should be sufficient time between adding the metrics to KAS Fleet Manager and the UI team to start working on switching to the new metrics so that there would be no problem with the metric history.  
+Because of this we should be able to expose the new metrics in KAS Fleet Manager and mark the existing metrics as deprecated as soon as they're configured to be pushed to Observatorium. 
+
+The existing metrics shall be marked as deprecated as per the MAS API deprecation policy.
+The deprecation will be communicated as follows:
+
+* The KAS Fleet Manager OpenAPI specification will document the deprecated metrics and their alternatives in each of the endpoint descriptions.
+* The metrics listed in the https://github.com/redhat-developer/app-services-guides/tree/main/docs/kafka/metrics-monitoring-kafka[Monitoring metrics in OpenShift Streams for Apache Kafka] documentation shall be marked as 'deprecated', pointing the users to use the newly revised metrics.
+* An email will be sent out to mas-devel and mas-all mailing list as per the deprecation policy.
+* Deprecation period shall be 60 days.
+** The metrics API is currently part of the KAS Fleet Manager API lifecycle tag.
+The KAS Fleet Manager API is in beta (no other tags are available at the moment).
+For beta, the deprecation period should be 60 days.
```

Comment:
Am I right in saying this? We haven't really specified what version our API is in. Should it be stable instead? If that's the case, the deprecation period should probably be 180 days. 

@pb82 @davidffrench

Reply:
@JameelB where does the beta tag come from? I'm not seeing it in our OpenAPI file.

Reply:
It was mentioned in the architecture call. See  https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/64#discussion_r914837142. This isnt in our openapi spec/api path. We never really specified  if our api is in beta/stable version.

Reply:
We could probably include this in the discussion of the api versioning though, so may be out of scope for this. 

For now, since 60 days was agreed on for the deprecation period of the old metrics we could state that here, different deprecation periods for beta/stable version could be something to think about in the future. wdyt?

Reply:
Agree, 60 days seems plenty. The UI only allows you to go back 7 days, so this would only affect customers using the federate endpoint.